toponetx/classes/path_complex.py

```diff
@@ -0,0 +1,502 @@
+"""Path complex."""
+from collections.abc import Hashable, Iterable, Iterator
+from itertools import chain
+from typing import List, Set, Tuple, Union
+
+import networkx as nx
+import numpy as np
+import scipy as sp
+
+from toponetx.classes.complex import Complex
+from toponetx.classes.path import Path
+from toponetx.classes.reportviews import NodeView, PathView
+
+__all__ = ["PathComplex"]
+
+
+class PathComplex(Complex):
+    """A class representing a path complex.
+
+    Class for constructing path complexes from graphs or iterables of paths.
+    The path complex is a simplicial complex if certain conditions are met (https://arxiv.org/pdf/1207.2834.pdf).
+    """
+
+    def __init__(
+        self,
+        paths=None,
+        name: str = "",
+        reserve_sequence_order: bool = False,
+        allowed_paths: List[Tuple] = None,
+        max_rank: int = 3,
+        **kwargs,
+    ) -> None:
+        """Class representing a path complex.
+
+        Parameters
+        ----------
+        paths : nx.Graph or iterable of paths
+            The paths in the path complex. If a graph is provided, the path complex will be constructed from the graph, and allowed paths are automatically computed.
+        name : str, optional
+            A name for the path complex.
+        reserve_sequence_order : bool, default=False
+            If True, reserve the order of the sub-sequence of nodes in the p-path. Else, the sub-sequence of nodes in the p-path will
+            be reversed if the first index is larger than the last index.
+        allowed_paths : List[Tuple], optional
+            A list of allowed boundaries. If None, only obvioys boundaries are constructed (sub-sequences where the first or the last index is omitted).
+        max_rank : int, default=3
+            The maximal length of a path in the path complex.
+        attr: keyword arguments, optional
+            Additional attributes to be associated with the path complex.
+        """
```

Comment:
Document the parameters in the class docstring, see other examples in this package.